lib/storage/src/content_manager/snapshots/mod.rs

```diff
@@ -54,15 +55,27 @@ pub async fn get_full_snapshot_path(
     Ok(absolute_snapshot_path)
 }
 
+pub async fn get_full_s3_snapshot_path(
+    toc: &TableOfContent,
+    snapshot_name: &str,
+) -> Result<PathBuf, StorageError> {
+    let absolute_snapshot_dir = PathBuf::from(toc.snapshots_path());
+    let absolute_snapshot_path = absolute_snapshot_dir.join(snapshot_name);
+    Ok(absolute_snapshot_path)
+}
+
 pub async fn do_delete_full_snapshot(
     dispatcher: &Dispatcher,
     access: Access,
     snapshot_name: &str,
 ) -> Result<JoinHandle<Result<bool, StorageError>>, StorageError> {
     access.check_global_access(AccessRequirements::new().manage())?;
     let toc = dispatcher.toc(&access);
-    let snapshot_manager = toc.get_snapshots_storage_manager();
-    let snapshot_dir = get_full_snapshot_path(toc, snapshot_name).await?;
+    let snapshot_manager = toc.get_snapshots_storage_manager().await;
+    let snapshot_dir = match snapshot_manager {
+        SnapshotStorageManager::LocalFS(_) => get_full_snapshot_path(toc, snapshot_name).await?,
+        SnapshotStorageManager::S3(_) => get_full_s3_snapshot_path(toc, snapshot_name).await?,
```

Comment:
`snapshot_manager` should have enough info to resolve this  internally.

Reply:
I initially considered this, but `get_full_snapshot_path` needs to be relocated under `snapshot_manager` to achieve this so I didn't do that. But yeah, I will definitely address this. : )

Perhaps I'll create a separate PR for this fix, as the current one is already quite large.